datalad/tests/testrepos/items.py

```diff
@@ -0,0 +1,1396 @@
+                # For additional testing of our datalad remote to not interfere
+                # and manage to handle all http urls and requests:
+                init_datalad_remote(AnnexRepo(self._path, init=False, create=False),
+                                    'datalad', autoenable=True)
+
+            if self._annex_direct:
+                annex_cmd = ['git', 'annex', 'direct']
+                _excute_by_item(cmd=annex_cmd, item=self,
+                                exc=TestRepoCreationError(
+                                    "Failed to switch to direct mode")
+                                )
+                # We just created the repo. Go for 'master'.
+                # TODO: This might be different if cloned
+                self._branches.add('annex/direct/master')
+
+            else:
+                # TODO: we didn't want direct mode (False) or didn't care (None).
+                # check whether it was enforced by annex and adjust attribute or
+                # raise
+                pass
+
+    def assert_intact(self):
+        """This supposed to make basic tests on whether or not what is stored in
+        this ItemRepo instance actually is, what can be found on disc plus some
+        consistency checks for the object itself.
+        Everything else is out of scope and needs to be tested by ItemRepo and
+        the subclasses of TestRepo.
+        """
+
+        log("Integrity check for %s(%s)", self.__class__, self.path)
+
+        # object consistency
+        if self.is_git:
+            assert(not self.is_annex)
+            assert(self.annex_is_initialized is False)
+            assert(self.annex_version is None)
+            assert(self.is_direct_mode is False)
+
+        if self.is_annex:
+            assert(not self.is_git)
+            if self.annex_version or self.is_direct_mode:
+                assert(self.annex_is_initialized is True)
+            if not self.annex_is_initialized:
+                # This needs to be a clone
+                assert(self._src)
+                assert(('origin', self._src) in self.remotes)
+            else:
+                # TODO: V6 adjusted branch
+                any(b == 'git-annex' or 'annex/direct' in b
+                    for b in self.branches)
+
+        if self._src:
+            # Note: self._src indicates that we cloned the repo from somewhere.
+            # Therefore we have 'origin'. Theoretically there could be an
+            # ItemCommand that removed that remote, but left self._src.
+            # If that happens, that ItemCommand probably should be adapted to
+            # also remove self._src.
+            assert(self.remotes)
+
+        assert_is_instance(self.branches, set)
+        [assert_is_instance(b, string_types) for b in self.branches]
+
+        if self.branches:
+            assert self.commits
+            # TODO: Not necessarily vice versa? Could be just detached HEAD, I guess
```

Comment:
make sure that it also asserts on the states of the branches

although git-annex branch might be tricky, since many git-annex operations just adjust the timestamp within the log of the git-annex although no content being dropped etc. E.g. 'annex fsck' would change git-annex branch.